### ISSUE TITLE
Convert OpenSSL.crypto.X509(Req) to x509.Certificate(SigningRequest)

### ISSUE THREAD

**Issue by kuba** [pyca/cryptography#2123]
I need to retrieve SANs from X509 certs as returned by pyOpenSSL's `Connection.get_peer_certificate()`. AFAIK `pyOpenSSL` doesn't support getting SANs, so it would be nice to have a public API that converts `OpenSSL.crypto.X509` into `cryptography.x509.Certificate`, thus allowing to use https://cryptography.io/en/latest/x509/#cryptography.x509.SubjectAlternativeName API.

For the record, I'm currently able to hack it with
```python
def _pyopenssl_to_cryptography(x509):
    if x509.__class__ is OpenSSL.crypto.X509:
        return openssl.x509._Certificate(openssl.backend, x509._x509)
    elif x509.__class__ is OpenSSL.crypto.X509Req:
        return openssl.x509._CertificateSigningRequest(
            openssl.backend, x509._req)
    else:
        raise TypeError('Unknown input type: {0}'.format(x509.__class__))

```

It would be nice to do it with CSRs also.

**Comment by reaperhulk** [comment]
I'm -1 on this. I'm uncomfortable exposing an officially blessed way to create objects using cffi objects since that's an implementation detail that could conceivably change in the future. I'd prefer to tell people that they should serialize the cert and load it back in. Slower, but guaranteed functional.

**Comment by alex** [comment]
I don't think it should be based on cffi objects, but a `to_cryptography` (bad name) method on `OpenSSL.crypto.X509` doesn't sound super unreasonable?

**Comment by reaperhulk** [comment]
If the method lives in pyopenssl that makes more sense to me. This issue should be reopened against the pyopenssl repo (or a PR proposed with the desired functionality) and it can be debated there :)

**Comment by xdesai** [comment]
In case someone like me finds this issue, before they find the actual work done for it. Alex did make the `to_cryptography` method https://github.com/pyca/pyopenssl/pull/640
Released in pyOpenSSL 17.1.0
